File: Structural/Lab13/Typescript/src/Entities/Decorators/real_estate_experience_decorator.ts
import { IRealEstateExperience } from "../../Interfaces/real_estate_experience";

export abstract class RealEstateExperienceDecorator implements IRealEstateExperience {
    protected _realEstate: IRealEstateExperience;
    constructor(realEstate: IRealEstateExperience) { 
        this._realEstate = realEstate;
    }   

    public customizeExperience(): String {
        return this._realEstate.customizeExperience();
    }

}

File: Structural/Lab10/C#/Entities/Decorators/RealEstateExperienceDecorator.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Interfaces;

namespace C_.Entities.Decorators
{
    public abstract class RealEstateExperienceDecorator : IRealEstateExperience
    {
        protected IRealEstateExperience _realEstate;

        public RealEstateExperienceDecorator(IRealEstateExperience realEstate)
        {
            _realEstate = realEstate;
        }

        public virtual string CustomizeExperience()
        {
            return _realEstate.CustomizeExperience();
        }
    }

}
